src/storage.rs

```diff
@@ -21,8 +21,9 @@ const LOCK_FILE: &str = "pearl.lock";
 ///
 #[derive(Debug)]
 pub struct Storage<K> {
+    initialized: Option<()>,
     config: Config,
-    active_blob: Box<Option<Blob<K>>>,
+    active_blob: Option<Box<Blob<K>>>,
     blobs: Vec<Blob<K>>,
```

Comment:
It is better to define `blobs` as immutable array. When a new blob added you just reallocate the array with additional space for that blob. In this way you can access `blobs` for read without synchronization.

Reply:
In this case I can clone `Vec<_>` on read